### ISSUE TITLE
Contao 4 compatibility problem

### ISSUE THREAD

**Issue by koertho** [isotope/core#1931]
When using Isotope with Contao 4.4., I get following warnings: 

> Warning: Declaration of Isotope\Model\TypeAgent::createModelFromDbResult(Database\Result $objResult) should be compatible with Contao\Model::createModelFromDbResult(Contao\Database\Result $objResult)

> Warning: Declaration of Isotope\Model\TypeAgent::createCollectionFromDbResult(Database\Result $objResult, $strTable = NULL) should be compatible with Contao\Model::createCollectionFromDbResult(Contao\Database\Result $objResult, $strTable)

This should easily be fixed by using the contao namespaces.

Tested with:
- Contao 4.4.17 and Isotope 2.4.6
- Contao 4.4.18 and Isotope 2.5.0

**Comment by Toflar** [comment]
No idea really.

**Comment by giacomodefra** [comment]
The latest version of isotope is compatible with Contao 4 ?

**Comment by aschempp** [comment]
Isotope is compatible with Contao 4. These are just warnings, they should not show up in production.

**Comment by m-vo** [comment]
Using the old Contao global namespaces also makes working with tools like phpstan cumbersome. :disappointed:  As soon as I'm using an isotope model for instance, the analyzer will break (because the globally namespaced classes cannot be found with regular autoloading). 

They are all over the place though. I just did a rough search/replace and there were > 200 affected files. So not sure if this would ever be changed / merged in case of a PR.

**Comment by aschempp** [comment]
If you use Composer, the Isotope files are being loaded by Composer. What's your problem exactly?

**Comment by akroii** [comment]
I'm getting the same message in frontend and backend after I switched the debug mode to 'on'
Contao 4.8.5
Isotope 2.6.4

```
ErrorException:
Warning: Declaration of Isotope\Model\TypeAgent::createModelFromDbResult(Database\Result $objResult) should be compatible with Contao\Model::createModelFromDbResult(Contao\Database\Result $objResult)

  at vendor/isotope/isotope-core/system/modules/isotope/library/Isotope/Model/TypeAgent.php:365
  at Symfony\Component\Debug\ErrorHandler->handleError(2, 'Declaration of Isotope\\Model\\TypeAgent::createModelFromDbResult(Database\\Result $objResult) should be compatible with Contao\\Model::createModelFromDbResult(Contao\\Database\\Result $objResult)', '/Applications/MAMP/htdocs/tps/vendor/isotope/isotope-core/system/modules/isotope/library/Isotope/Model/TypeAgent.php', 365, array('class' => 'Isotope\\Model\\TypeAgent', 'e' => 8183, 'file' => '/Applications/MAMP/htdocs/tps/vendor/composer/../isotope/isotope-core/system/modules/isotope/library/Isotope/Model/TypeAgent.php'))
     (vendor/symfony/debug/DebugClassLoader.php:163)
  at spl_autoload_call('Isotope\\Model\\TypeAgent')
     (vendor/isotope/isotope-core/system/modules/isotope/library/Isotope/Model/Product.php:51)
  at spl_autoload_call('Isotope\\Model\\Product')
     (vendor/isotope/isotope-core/system/modules/isotope/config/config.php:24)
  at include('/Applications/MAMP/htdocs/tps/vendor/isotope/isotope-core/system/modules/isotope/config/config.php')
     (vendor/contao/core-bundle/src/Resources/contao/library/Contao/Config.php:148)
  at Contao\Config->initialize()
     (vendor/contao/core-bundle/src/Resources/contao/library/Contao/Config.php:113)
  at Contao\Config::getInstance()
     (vendor/contao/core-bundle/src/Framework/Adapter.php:38)
  at Contao\CoreBundle\Framework\Adapter->__call('getInstance', array())
     (vendor/contao/core-bundle/src/Framework/ContaoFramework.php:265)
  at Contao\CoreBundle\Framework\ContaoFramework->initializeFramework()
     (vendor/contao/core-bundle/src/Framework/ContaoFramework.php:127)
  at Contao\CoreBundle\Framework\ContaoFramework->initialize(true)
     (vendor/numero2/contao-marketing-suite/src/Routing/LinkShortenerRouteProvider.php:49)
  at numero2\MarketingSuiteBundle\Routing\LinkShortenerRouteProvider->getRouteCollectionForRequest(object(Request))
     (vendor/symfony-cmf/routing/src/NestedMatcher/NestedMatcher.php:141)
  at Symfony\Cmf\Component\Routing\NestedMatcher\NestedMatcher->matchRequest(object(Request))
     (vendor/symfony-cmf/routing/src/DynamicRouter.php:271)
  at Symfony\Cmf\Component\Routing\DynamicRouter->matchRequest(object(Request))
     (vendor/symfony-cmf/routing/src/ChainRouter.php:188)
  at Symfony\Cmf\Component\Routing\ChainRouter->doMatch('/software/zusatzprogramme/hs-finanzwesen', object(Request))
     (vendor/symfony-cmf/routing/src/ChainRouter.php:158)
  at Symfony\Cmf\Component\Routing\ChainRouter->matchRequest(object(Request))
     (vendor/symfony/http-kernel/EventListener/RouterListener.php:114)
  at Symfony\Component\HttpKernel\EventListener\RouterListener->onKernelRequest(object(RequestEvent), 'kernel.request', object(EventDispatcher))
     (vendor/symfony/event-dispatcher/EventDispatcher.php:298)
  at Symfony\Component\EventDispatcher\EventDispatcher::Symfony\Component\EventDispatcher\{closure}(object(RequestEvent), 'kernel.request', object(EventDispatcher))
     (vendor/symfony/event-dispatcher/EventDispatcher.php:260)
  at Symfony\Component\EventDispatcher\EventDispatcher->doDispatch(array(object(Closure), object(Closure), object(Closure), object(Closure), object(Closure), object(Closure), object(Closure), object(Closure), object(Closure), object(Closure), object(Closure), object(Closure), object(Closure), object(Closure), object(Closure), object(Closure), object(Closure), object(Closure), object(Closure), object(Closure), object(Closure), object(Closure), object(Closure), object(Closure), object(Closure)), 'kernel.request', object(RequestEvent))
     (vendor/symfony/event-dispatcher/EventDispatcher.php:235)
  at Symfony\Component\EventDispatcher\EventDispatcher->callListeners(array(object(Closure), object(Closure), object(Closure), object(Closure), object(Closure), object(Closure), object(Closure), object(Closure), object(Closure), object(Closure), object(Closure), object(Closure), object(Closure), object(Closure), object(Closure), object(Closure), object(Closure), object(Closure), object(Closure), object(Closure), object(Closure), object(Closure), object(Closure), object(Closure), object(Closure)), 'kernel.request', object(RequestEvent))
     (vendor/symfony/event-dispatcher/EventDispatcher.php:73)
  at Symfony\Component\EventDispatcher\EventDispatcher->dispatch(object(RequestEvent), 'kernel.request')
     (vendor/symfony/http-kernel/HttpKernel.php:127)
  at Symfony\Component\HttpKernel\HttpKernel->handleRaw(object(Request), 1)
     (vendor/symfony/http-kernel/HttpKernel.php:68)
  at Symfony\Component\HttpKernel\HttpKernel->handle(object(Request), 1, true)
     (vendor/symfony/http-kernel/Kernel.php:198)
  at Symfony\Component\HttpKernel\Kernel->handle(object(Request))
     (web/index.php:31)
```

**Comment by fritzmg** [comment]
Temporary patch solution for now:

1. Create a file called `patches/TypeAgent.patch` with the following content:

    ```diff
    --- a/vendor/isotope/isotope-core/system/modules/isotope/library/Isotope/Model/TypeAgent.php
    +++ b/vendor/isotope/isotope-core/system/modules/isotope/library/Isotope/Model/TypeAgent.php
    @@ -249,7 +249,7 @@
          *
          * @return \Model
          */
    -    public static function createModelFromDbResult(\Database\Result $objResult)
    +    public static function createModelFromDbResult(\Contao\Database\Result $objResult)
    {
          $strClass = '';
    
    @@ -297,7 +297,7 @@
          *
          * @return \Model\Collection
          */
    -    protected static function createCollectionFromDbResult(\Database\Result $objResult, $strTable = null)
    +    protected static function createCollectionFromDbResult(\Contao\Database\Result $objResult, $strTable = null)
    {
          // @deprecated Remove in Isotope 3.0 (only for backward compatibility Isotope < 2.1.2)
          if (null === $strTable) {
    ```

2. Insert the following into the `"extra"` part of your `composer.json`:

    ```json
    "patches": {
        "isotope/isotope-core": [
            "patches/TypeAgent.patch"
        ]
    }
    ```

3. Execute the following command:

    ```
    composer require cweagans/composer-patches:*
    ```

This will apply the patch from now on.

**Comment by aschempp** [comment]
fixed in d8e6df3aefb49b654de1b72304e55fa389446481
